calicoctl/commands/clientmgr/client.go

```diff
@@ -30,6 +33,37 @@ import (
 // parameter not loaded from file.
 func NewClient(cf string) (client.Interface, error) {
 	cfg, err := LoadClientConfig(cf)
+
+	// Set kubeconfig path
+	kubeconfigPath, exists := os.LookupEnv("KUBECONFIG")
+	if !exists {
+		kubeconfigPath = filepath.Join(
+			os.Getenv("HOME"), ".kube", "config",
+		)
+	}
+	cfg.Spec.KubeConfig.Kubeconfig = kubeconfigPath
+
+	// Look up datastore type
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
```

Comment:
IIUC this will fail if the kubeconfig does not exist, and therefore this is changing the behavior of calicoctl such that it will only be useable if a kubeconfig is present. I don't think that is correct.  If the kubeconfig is not present then we should just pull the environments from the host - this is already what happens in LoadClientConfig() at line 35, so I think if the kubeconfig does not exist then we just use the original client config.